File: server/src/main/java/com/thxran/dropbox/entity/File.java
package com.thxran.dropbox.entity;

import com.thxran.dropbox.utils.CUID;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "files")
@EntityListeners(AuditingEntityListener.class)
public class File {
    @Id
    private String id;
    private String fileName;
    private String fileType;
    private String fileSize;
    private String fileStoragePath;

    @Column(nullable = false)
    private String fileURL;

    private boolean isArchived = false;
    private boolean isFavorite = false;

    @Column(nullable = false)
    private String folderId;

    @Column(nullable = false)
    private String userId;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void generateId() {
        if (id == null) {
            id = CUID.generateCUID();
        }
    }
}

File: client/schema/file.ts
import { z } from "zod";

export const uploadSchema = z.object({
  fileName: z.string(),
  fileType: z.string(),
  fileSize: z.string(),
  fileStoragePath: z.string(),
  fileURL: z.string(),
  folderId: z.string(),
  userId: z.optional(z.string()),
});

export const renameFile = z.object({
  newname: z
    .string()
    .min(2, {
      message: "min of 2 characters required!",
    })
    .max(20, {
      message: "max of 20 characters!",
    }),
});
